### DISCUSSION TITLE
Bulk Data Import Design Best Practices?

### DISCUSSION THREAD

**Discussion by jme783** [twilio-labs/paste#754] [Office hours] [answered]
Hello there,

I'm designing a UI that involves bulk importing of many records at the same time. My thought is that a file-based upload (e.g. CSV) would be most appropriate. I wanted to ask if there were any design standards or best practices that I could pull from to accomplish this. I haven't come across this kind of interaction in the Twilio Console or in the Paste docs. The closest thing to what I think I'm looking for is something like what [Flatfile](https://flatfile.io/) offers.

Thanks in advance for the help!

**Comment by serifluous** [comment, marked as answer]
Hey @jme783 ! What info are you trying to import in each record?

I'm going to reach out to some product designers, but you could take a look at this SendGrid flow for importing Contacts: https://mc.sendgrid.com/contacts/import/upload-csv

^ You can play around with the interaction if you have an account, download the sample CSV from that page, and reupload it.

![Screen Shot 2020-09-11 at 10 53 01 AM](https://user-images.githubusercontent.com/3821840/92958054-6e99f080-f41e-11ea-9549-0b6e395ab842.png)
![Screen Shot 2020-09-11 at 11 00 14 AM](https://user-images.githubusercontent.com/3821840/92958071-73f73b00-f41e-11ea-8431-b3cd9f2862d9.png)

**Reply by serifluous** [reply]
Asked around (thanks, @elainelylee!) and was pointed to a similar interaction in Assets in Twilio Console in both Assets (Classic) and when you're in a Service and upload files.

**Comment by jme783** [comment]
This is super helpful @serifluous thank you so much

**Reply by serifluous** [reply]
You're welcome! If you want to get some feedback on a prototype, we totally welcome you at our office hours/crit. We can invite some product designers who worked on the SendGrid and Console Assets flows to give some insight, too.
